src/flint/FlintTypes.jl

```diff
@@ -2100,13 +2100,22 @@ end
 
    var::String
    
-   overfields :: Dict{Int, Vector{FinFieldMorphism}}
-   subfields :: Dict{Int, Vector{FinFieldMorphism}}
+   overfields::Dict{Int, Vector{FinFieldMorphism}}
+   subfields::Dict{Int, Vector{FinFieldMorphism}}
+
+   isabsolute::Bool
+   base_field
+   defining_poly
```

Comment:
other than not putting a type here, I don't see anything wrong with this pr

Reply:
I am going to add a few mores tests though

Reply:
I don't think we can add forward references to types defined later.

If you are happy with it, I will do a bit of clean up.

I added a question to my first post. What is your opinion?

Reply:
Oh, well, I don't think caching (with a nullptr, i.e. !isdefined) it will work because inplace operations would have to clear the cached representation and you can't change it from a non-nullptr back to a nullptr.

Reply:
Ah good point. I might go down the `Union{Nothing,...}` route. I don't want `[coeff(x, i) for i in 0:degree(F)]` to be too slow.